bundle/bundle.go

```diff
@@ -0,0 +1,88 @@
+// Package bundle implements a SPIFFE-compliant bundle type.
+package bundle
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/zeebo/errs"
+	"gopkg.in/square/go-jose.v2"
+)
+
+const (
+	x509SVIDUse = "x509-svid"
+	jwtSVIDUse  = "jwt-svid"
+)
+
+// Bundle is a SPIFFE bundle object. It follows the definitions given in the
+// SPIFFE standard:
+//
+// https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE_Trust_Domain_and_Bundle.md#4-spiffe-bundle-format
+type Bundle struct {
+	// SPIFFE bundles are represented as an RFC 7517 compliant JWK Set.
+	jose.JSONWebKeySet
+
+	// Sequence is a monotonically increasing integer that is changed whenever
+	// the contents of the bundle are updated.
+	Sequence uint64 `json:"spiffe_sequence,omitempty"`
+
+	// RefreshHint indicates how often a consumer should check back for updates
+	// (in seconds).
+	RefreshHint int `json:"spiffe_refresh_hint,omitempty"`
+}
+
+// SPIFFEInvalidJWK is used to describe why a key is not SPIFFE-compliant.
+type SPIFFEInvalidJWK struct {
+	jose.JSONWebKey
+	reason error
+}
+
+// Decode reads a json document from a Reader interface and turns it into a
+// bundle object.
+func Decode(r io.Reader) (*Bundle, error) {
+	doc := new(Bundle)
+	if err := json.NewDecoder(r).Decode(doc); err != nil {
+		return nil, fmt.Errorf("failed to decode bundle: %v", err)
+	}
+
```

Comment:
I think we should be safe by default and call `ValidateKeys`. If people want to be unsafe, they can always unmarshal it directly.

I also think this function should fail if there are any invalid keys. We can then provide another method (e.g. `DecodeLenient`) that returns the invalid keys separately (after pruning them from the bundle)
```suggestion
func DecodeLenient(r io.Reader) (*Bundle, []InvalidKey, error)
```